### PR TITLE
Add e2e test for sanitizers never error

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Add k8s e2e test for https://github.com/solo-io/gloo/pull/6289#pullrequestreview-938372272 , basically equivalent to these steps: https://github.com/solo-io/gloo/issues/6115#issuecomment-1091856972

also ran the new test locally against gloo edge 1.12.0-beta2 and confirmed it fails before the changes aforementioned, and passes on version 1.12.0-beta3

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/gateway/robustness_test.go [hunk 1]
```diff
@@ -292,6 +329,134 @@ var _ = Describe("Robustness tests", func() {
...
+
+				// can no longer route to appName2 since its k8s service has been removed
+				Eventually(func() (string, error) {
+					return testHelper.Curl(secondRouteCurlOpts())
+				}, "30s", "1s").Should(BeEmpty())
+				Consistently(func() (string, error) {
+					return testHelper.Curl(secondRouteCurlOpts())
+				}, "5s", "1s").Should(BeEmpty())
+
+				// roll pods to ensure we are resilient to pod restarts
```
Comment (bewebi) [inline test/kube2e/gateway/robustness_test.go:436]: What makes pod restarts an interesting case to test?
If it's conceivable that the "works with deleted services" behavior will be good but "works with deleted services after pod restart(s)" won't, we should consider having a separate `It()` for the latter

Comment (kdorosh) [reply test/kube2e/gateway/robustness_test.go:436]: if we have invalid config we may not update our xds cache that gloo is serving (this was the old bug)

without rolling the pods envoy will just keep the old config; but we need to validate that even after pods roll that it is actively getting good config still

the test before this one checks for endpoint updates without a pod roll which seems to cover the other desired case

Comment (bewebi) [reply test/kube2e/gateway/robustness_test.go:436]: Discussed offline, it's worth the quick add of a test that does everything up to this point as a simpler case that does not test resilience to pod restarts since that behavior could be correct even if were not resilient to pod restarts
This was effectively the case prior to the change we're testing

Comment (kdorosh) [reply test/kube2e/gateway/robustness_test.go:454]: addressed in https://github.com/solo-io/gloo/pull/6303/commits/6ea3b4744c1cc90a0548d702854bbce5fd45e04d

**Review by jenshu — APPROVED** [review]
